### PR TITLE
storage: Refactor mount user handling

### PR DESCRIPTION
**Pull request by mvollmer**
Finding and stopping processes that use a mount point started out as a
feature for NFS only but is now used across all of the Storage page.
Let's reflect that in the code.

### CODE REVIEW

File: pkg/storaged/client.js [hunk 1]
```diff
@@ -602,6 +603,24 @@ client.older_than = function older_than(version) {
...
+
+client.find_mount_users = (target, is_mounted) => {
+    if (is_mounted === undefined || is_mounted)
+        return python.spawn(mount_users_py, ["users", target], { superuser: "try", err: "message" }).then(JSON.parse);
+    else
+        return Promise.resolve([]);
+};
+
+client.stop_mount_users = (users) => {
+    if (users && users.length > 0) {
```
Comment (cockpituous) [inline pkg/storaged/client.js:617]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17670-20220825-111050-b8a78922-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/client.js [hunk 2]
```diff
@@ -602,6 +603,24 @@ client.older_than = function older_than(version) {
...
+        return python.spawn(mount_users_py, ["stop", JSON.stringify(users)],
+                            { superuser: "try", err: "message" });
+    } else
+        return Promise.resolve();
```
Comment (cockpituous) [inline pkg/storaged/client.js:621]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17670-20220825-111050-b8a78922-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/client.js [hunk 3]
```diff
@@ -694,29 +713,16 @@ function nfs_mounts() {
...
-            JSON.stringify(entry)]);
+        return client.stop_mount_users(users).then(() => unmount_entry(entry));
     }
 
     function stop_and_remove_entry(users, entry) {
-        const [pids, units] = pids_and_units(users);
-        return spawn_nfs_mounts(["stop-and-remove",
-            JSON.stringify(units), JSON.stringify(pids),
-            JSON.stringify(entry)]);
+        return client.stop_mount_users(users).then(() => remove_entry(entry));
```
Comment (cockpituous) [inline pkg/storaged/client.js:721]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17670-20220825-111050-b8a78922-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/client.js (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/client.js:617]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17670-20220825-201517-e2bf9819-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/client.js (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/client.js:621]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17670-20220825-201517-e2bf9819-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**